### ISSUE TITLE
Image storing in original but not new image

### ISSUE THREAD

**Issue by Lunteer** [ysugimoto/aws-lambda-image#174]
When uploading an image to s3 the image is copied to the /original/ folder but the new image is not written.  I get an accessed denied. Below is the log report and the permissions for the IAM. Do I need to set something on the bucket itself?

```
2019-02-14T14:08:22.903Z d2bfd419-6eb6-4580-9094-ad1fc36387e7 Parsing S3 event...
2019-02-14T14:08:22.926Z d2bfd419-6eb6-4580-9094-ad1fc36387e7 Downloading: example.jpg
2019-02-14T14:08:23.109Z d2bfd419-6eb6-4580-9094-ad1fc36387e7 Backing up to: ./original
2019-02-14T14:08:23.110Z d2bfd419-6eb6-4580-9094-ad1fc36387e7 Uploading to: original/example.jpg (121569 bytes)
2019-02-14T14:08:23.204Z d2bfd419-6eb6-4580-9094-ad1fc36387e7 Reducing to: ./
2019-02-14T14:08:23.525Z d2bfd419-6eb6-4580-9094-ad1fc36387e7 Uploading to: reduced-example.jpg (151811 bytes)
2019-02-14T14:08:23.577Z d2bfd419-6eb6-4580-9094-ad1fc36387e7
{
    "errorMessage": "Error processing example.jpg: AccessDenied: Access Denied"
}
END RequestId: d2bfd419-6eb6-4580-9094-ad1fc36387e7
```
Here are the permissions - changed some references to generic to not give out sensitive data

**Function policy**
```
{
  "Version": "2012-10-17",
  "Id": "default",
  "Statement": [
    {
      "Sid": "bucket_name-access-XXXXXX",
      "Effect": "Allow",
      "Principal": {
        "Service": "s3.amazonaws.com"
      },
      "Action": "lambda:InvokeFunction",
      "Resource": "arn:aws:lambda:us-east-1:XXXXXX:function:name",
      "Condition": {
        "ArnLike": {
          "AWS:SourceArn": "arn:aws:s3:::bucket_name"
        }
      }
    },
    {
      "Sid": "bucket_name-access-XXXXXXX",
      "Effect": "Allow",
      "Principal": {
        "Service": "s3.amazonaws.com"
      },
      "Action": "lambda:InvokeFunction",
      "Resource": "arn:aws:lambda:us-east-1:XXXXXXXXXXXX:function:name",
      "Condition": {
        "ArnLike": {
          "AWS:SourceArn": "arn:aws:s3:::bucket_name"
        }
      }
    },
    {
      "Sid": "bucket_name-access-XXXXXXXXX",
      "Effect": "Allow",
      "Principal": {
        "Service": "s3.amazonaws.com"
      },
      "Action": "lambda:InvokeFunction",
      "Resource": "arn:aws:lambda:us-east-1:XXXXXX:function:name",
      "Condition": {
        "ArnLike": {
          "AWS:SourceArn": "arn:aws:s3:::bucket_name"
        }
      }
    },
    {
      "Sid": "bucket_name-access-XXXXXXXXXX",
      "Effect": "Allow",
      "Principal": {
        "Service": "s3.amazonaws.com"
      },
      "Action": "lambda:InvokeFunction",
      "Resource": "arn:aws:lambda:us-east-1:XXXXXXXXXX:function:name",
      "Condition": {
        "ArnLike": {
          "AWS:SourceArn": "arn:aws:s3:::bucket_name"
        }
      }
    }
  ]
}
```
**Execution role**
```
{
  "roleName": "name",
  "policies": [
    {
      "document": {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
              "lambda:CreateFunction",
              "s3:GetObjectVersionTagging",
              "lambda:UpdateFunctionConfiguration",
              "s3:PutObjectVersionTagging",
              "lambda:GetFunctionConfiguration",
              "lambda:UpdateAlias",
              "lambda:UpdateFunctionCode",
              "s3:PutObject",
              "s3:GetObjectAcl",
              "s3:GetObject",
              "lambda:AddPermission",
              "s3:PutObjectVersionAcl",
              "s3:GetObjectVersionAcl",
              "s3:GetObjectTagging",
              "s3:PutObjectTagging",
              "s3:GetObjectVersionForReplication",
              "s3:PutObjectAcl",
              "s3:GetObjectVersion",
              "lambda:CreateAlias"
            ],
            "Resource": [
              "arn:aws:lambda:us-east-1:XXXXXXXXXXX:function:name",
              "arn:aws:s3:::bucket_name/*"
            ]
          }
        ]
      },
      "name": "name",
      "id": "XXXXXXXXXXXXXXXXXXX",
      "type": "managed",
      "arn": "arn:aws:iam::XXXXXXXXXX:policy/name"
    },
    {
      "document": {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "logs:CreateLogGroup",
              "logs:CreateLogStream",
              "logs:PutLogEvents"
            ],
            "Resource": "*"
          }
        ]
      },
      "name": "AWSLambdaBasicExecutionRole",
      "id": "XXXXXXXXXXXXXXX",
      "type": "managed",
      "arn": "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
    }
  ],
  "trustedEntities": [
    "lambda.amazonaws.com"
  ]
}
```
Thanks in advance

**Comment by Lunteer** [comment]
So I tried to change around bucket policies and did manage to get the new processed image to write into the S3 bucket by changing the Public Access Setting of "Block new public ACLs and uploading public objects" to False.

Obviously this is not ideal so want to change the IAM role/user/function (not even sure which one) to the equivalent.

Any help would be much appreciated.

**Comment by Lunteer** [comment]
After on and off testing for about 2 weeks, finally found the issue. The acl was set to 'public-read' and this what was blocking the image from being written in the s3 bucket. Changed the acl to 'private' and it is now uploading the new reduced image. Interestingly, I did read about this issue and added 's3:PutObjectAcl' to the role and still it did not help. This is what I am now trying to understand.

Here is the [thread](https://stackoverflow.com/questions/34049794/aws-lambda-with-administratoraccess-throws-access-denied-when-saving-to-s3-wit) that broke that lead to this discovery.
